src/main/kotlin/no/nav/pensjon/kalkulator/tjenestepensjonsimulering/api/map/TjenestepensjonSimuleringSpecMapperV2.kt

```diff
@@ -0,0 +1,122 @@
+                        spec.foedselsdato,
+                        Alder(spec.gradertUttak.uttaksalder.aar, spec.gradertUttak.uttaksalder.maaneder)
+                    ),
+                    beloep = it
+                )
+            )
+        }
+
+        spec.heltUttak.aarligInntektVsaPensjon?.let {
+            inntekter.add(
+                FremtidigInntektV2(
+                    fom = uttakDato(
+                        spec.foedselsdato,
+                        Alder(spec.heltUttak.uttaksalder.aar, spec.heltUttak.uttaksalder.maaneder)
+                    ),
+                    beloep = it.beloep
+                )
+            )
+            inntekter.add(
+                FremtidigInntektV2(
+                    fom = uttakDato(
+                        spec.foedselsdato,
+                        Alder(it.sluttAlder.aar, it.sluttAlder.maaneder)
+                    ).plusMonths(1), //sluttAlder er tom, '0' inntekt starter en måned etter
+                    beloep = 0
+                )
+            )
+        } ?: inntekter.add(
+            FremtidigInntektV2(
+                fom = uttakDato(
+                    spec.foedselsdato,
+                    if (inntekter.isEmpty()){
+                        mapToUttaksalder(spec) //0-inntekt starter ved gradert uttaksdato med mindre helt uttak er spesifisert, da starter 0-inntekt ved helt uttaksdato
+                    }
+                    else { // gradert uttak er spesifisert med inntekt ved siden av, 0-inntekt starter ved helt uttaksdato
+                        Alder(spec.heltUttak.uttaksalder.aar, spec.heltUttak.uttaksalder.maaneder)
+                    }
+                ),
+                beloep = 0
+            )
+        )
+
+        return inntekter
+    }
+
+    private fun mapToUttaksalder(spec: IngressSimuleringOffentligTjenestepensjonSpecV2): Alder {
+        return (spec.gradertUttak?.uttaksalder ?: spec.heltUttak.uttaksalder)
+            .let { Alder(it.aar, it.maaneder) }
+    }
+
+    private fun antallAar(oppholdListe: List<UtenlandsoppholdV2>): Int {
+        val sammenslattePerioder = slaaSammenOverlappendePerioder(oppholdListe)
+        val antallDager = antallDager(sammenslattePerioder)
+        return (antallDager / DAGER_PER_AAR).toInt()
+    }
+
+    private fun antallDager(oppholdListe: List<UtenlandsoppholdV2>): Long {
+        return oppholdListe.sumOf { (it.tom ?: LocalDate.now()).toEpochDay() + 1 - it.fom.toEpochDay() }
+    }
+
+    private fun slaaSammenOverlappendePerioder(oppholdListe: List<UtenlandsoppholdV2>): List<UtenlandsoppholdV2> {
+        if (oppholdListe.isEmpty()) return emptyList()
+        val sortertePerioder = oppholdListe.sortedBy { it.fom }
+
+        val sammenslaatte = mutableListOf<UtenlandsoppholdV2>()
+        var gjeldendePeriode = sortertePerioder.first()
+
+        for (periode in sortertePerioder.drop(1)) {
+            if (periode.fom <= (gjeldendePeriode.tom ?: LocalDate.now()).plusDays(1)) {
```

Comment:
Hva skjer hvis `for`-løkka startes et nanosekund før midnatt? Da vil `LocalDate.now()` potensielt endres fra en iterasjon til den neste. Litt søkt, men det kan vel hende?